### DISCUSSION TITLE
how to include missing commits in metrics?

### DISCUSSION THREAD

**Discussion by GalaxySH** [lowlighter/metrics#381] [Q&A] [answered]
Just looking at the output of the languages section, I can tell there are many commits and lines missing. I am sure this has something to do with commit authoring under different credentials, but I'm not sure how to fix this.

Additionally, would I be able to include commits in my organization account?

**Comment by lowlighter** [comment, marked as answer]
Commits you made are detected through `git log --author=<commits_authoring> --regexp-ignore-case`, so basically you need to find under which email you committed (login and names may work too though you could have collisions). This option is a comma separated array, so `commits_authoring: GalaxySH, GalaxySH@users.noreply.github.com, GalaxySH@company.com` should work if you have multiple addresses or names

To change scopes of repositories affected, you may also need to tweak with [`repositories_affiliations`](https://github.com/lowlighter/metrics/blob/master/source/plugins/base/metadata.yml#L40-L48) option 

Huge commits may not be processable so that's why the number of commits used to compute stats was added.
It'll help detecting whether some commits have been skipped or not (I'm not sure of the root cause, but it's probably due to memory issues)

**Reply by GalaxySH** [reply]
> commits_authoring: GalaxySH, GalaxySH@users.noreply.github.com, GalaxySH@company.com

I had already tried to do that with commit authoring, but I looked up the author email of my latest commits and tried it again. I also changed the `repositories_affiliations` to `collaborator`.

When the metrics updated, my line count actually got even smaller.

**Reply by lowlighter** [reply]
I think the broadest affiliations is the empy one, I suspect that `collaborator` in GitHub api actually means that you have push access to repository so that's probably why it got smaller

Your `commits_authoring` seems fine.

I don't remember whether there were some improves between 3.10 (@latest) and 3.11 (@master) but maybe you'll get better results on current developpement branch? At least it'll tell how many commits were used and how much were skipped

**Reply by GalaxySH** [reply]
I left the affiliations setting where it was and switched to the master branch, and now it works. I don't know what changed but it seems to be counting correctly, from what I saw at first glance.
